tests/v2/validation/Jenkinsfile_support_matrix_os

```diff
@@ -0,0 +1,210 @@
+                          string(credentialsId: 'RANCHER_LINODE_ACCESSKEY', variable: "RANCHER_LINODE_ACCESSKEY")]) {
+          
+        withEnv(paramsMap) {
+          stage('Checkout') {
+            deleteDir()
+            checkout([
+                      $class: 'GitSCM',
+                      branches: [[name: "*/${branch}"]],
+                      extensions: scm.extensions + [[$class: 'CleanCheckout']],
+                      userRemoteConfigs: scm.userRemoteConfigs
+                    ])
+          }
+          dir ("./") {
+              stage('Configure and Build') {
+                if (env.AWS_SSH_PEM_KEY && env.AWS_SSH_KEY_NAME) {
+                  dir("./tests/v2/validation/.ssh") {
+                    def decoded = new String(AWS_SSH_PEM_KEY.decodeBase64())
+                    writeFile file: AWS_SSH_KEY_NAME, text: decoded
+                  }
+                }
+
+                sh "./tests/v2/validation/configure.sh"
+
+              }
+              stage('Setup') {
+                
+              } 
+              stage('Run Jobs') {
+                dir("./tests/v2/validation/") {
+                  // install yq
+                  sh returnStdout: true, script: 'wget -qO ./yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64'
+                  sh returnStdout: true, script: 'chmod a+x ./yq'
+                  def amis = env.RANCHER_AMI_LIST
+                  amiList = amis.split(',').each { obj -> 
+                  String ami
+                  String sshUser
+                  if (obj.contains(':')){
+                    def data = obj.split(':')
+                    ami = data[0]
+                    sshUser = data[1]
+                  } else {
+                    ami = obj
+                    sshUser = 'ec2-user'
+                  }
+
+                  env.RKE1_NODE_CONFIG = sh (
+                  script: "./yq '.awsNodeTemplate.ami = \"$ami\"'",
+                  returnStdout: true
+                  ).trim()
+
+                  env.RKE1_NODE_CONFIG = sh (
+                  script: "./yq '.awsNodeTemplate.sshUser = \"$sshUser\"'",
+                  returnStdout: true
+                  ).trim()
+
+                  env.RKE2_NODE_CONFIG = sh (
+                  script: "./yq '.awsMachineConfig.ami = \"$ami\"'",
+                  returnStdout: true
+                  ).trim()
+
+                  env.RKE2_NODE_CONFIG = sh (
+                  script: "./yq '.awsMachineConfig.sshUser = \"$sshUser\"'",
+                  returnStdout: true
+                  ).trim()
+
+                  // RKE1customconfig
+
+                  def rke1CustomFilename = "rke1CustomConfig.yaml"
+                  def rke1CustomConfig = env.RANCHER_RKE1_CUSTOM_CONFIG
+
+                  writeFile file: rke1CustomFilename, text: rke1CustomConfig
+
+                  sh "./yq e '.awsEC2Configs.awsEC2Config.[0].awsAMI = \"$ami\"' -i rke1CustomConfig.yaml"
```

Comment:
You need to do the same for nodeTemplate and machineConfig

Reply:
all the configs are now written to files like this